### PR TITLE
Configure batches with estimated run times

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: jade/cli/submit_jobs.py [hunk 1]
```diff
@@ -29,7 +29,8 @@
     "-b", "--per-node-batch-size",
     default=DEFAULTS["per_node_batch_size"],
     show_default=True,
-    help="Number of jobs to run on one node in one batch."
+    help="Number of jobs to run on one node in one batch. If 0, auto-determine batches based on "
```
Comment (daniel-thom) [inline jade/cli/submit_jobs.py:32]: This was the simplest implementation but is not necessarily the most intuitive for users. I'm open to suggestions.

File: jade/cli/submit_jobs.py [hunk 2]
```diff
@@ -150,6 +160,16 @@ def submit_jobs(
             sys.exit(1)
         hpc_config = HpcConfig(**load_data(hpc_config))
 
+    if time_based_batching and per_node_batch_size != DEFAULTS["per_node_batch_size"]:
+        # This doesn't catch the case where the user passes --per-node-batch-size=default, but
+        # I don't see that click provides a way to detect that condition.
```
Comment (jgu2) [inline jade/cli/submit_jobs.py:165]: Not sure if it worths the effort, another idea with `click` is using `User Input Prompts` for this feature, and ensure user's inputs with interactive session.

File: jade/hpc/hpc_submitter.py [hunk 3]
```diff
@@ -108,16 +105,16 @@ def run(self):
...
-            if batch.is_job_blocked(job, try_add_blocked_jobs):
+            if batch.is_job_blocked(job):
                 blocked_jobs.append(job)
             else:
                 jade_job.set_blocking_jobs(job.blocked_by)
                 batch.append(jade_job)
                 submitted_jobs.append(job)
 
-            if batch is not None and batch.num_jobs >= self._params.per_node_batch_size:
+            if batch is not None and batch.ready_to_submit():
```
Comment (jgu2) [inline jade/hpc/hpc_submitter.py:117]: May remove `batch is not None`?  lines 107-108 ensured `batch`.

**Review by jgu2 — APPROVED** [review]
Tests work good.

File: jade/cli/submit_jobs.py (hunk 2 above)
Comment (daniel-thom) [reply jade/cli/submit_jobs.py:165]: Good suggestion, but I will defer. We could probably make lots of changes like that.

**Pull request merged**